src/AutoRest.CSharp/Common/Output/Builders/XmlSerializationMethodsBuilder.cs

```diff
@@ -215,11 +255,20 @@ private static IEnumerable<MethodBodyStatement> BuildDeserializeBody(XElementExp
 
             var objectType = (ObjectType)objectSerialization.Type.Implementation;
             var parameterValues = propertyVariables.ToDictionary(v => v.Key.SerializationConstructorParameterName, v => (ValueExpression)v.Value);
-            var parameters = objectType.SerializationConstructor.Signature.Parameters
-                .Select(p => parameterValues[p.Name])
-                .ToArray();
 
-            yield return Return(New.Instance(objectSerialization.Type, parameters));
+            var arguments = new List<ValueExpression>();
+            foreach (var parameter in objectType.SerializationConstructor.Signature.Parameters)
+            {
+                if (parameterValues.TryGetValue(parameter.Name, out var argument))
+                    arguments.Add(argument);
+                else
+                {
+                    // this must be the raw data property
+                    arguments.Add(new PositionalParameterReference(parameter.Name, Null));
+                }
+            }
+
+            yield return Return(New.Instance(objectSerialization.Type, arguments.ToArray()));
```

Comment:
have we considered to make an overload to take in `IReadOnlyList<>` to void `ToArray` of making another copy?